db_migrate/versions/dd9c97ead24_share_the_locking_of_runs.py

```diff
@@ -0,0 +1,35 @@
+"""Share the locking of runs across servers via database
+
+Revision ID: dd9c97ead24
+Revises: 4b38fa14c27b
+Create Date: 2017-11-17 15:44:07.810579
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = 'dd9c97ead24'
+down_revision = '4b38fa14c27b'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+
+from libcodechecker.server.database.run_db_model import Run
+
+
+def upgrade():
+    op.add_column('runs', sa.Column('lock_timestamp', sa.DateTime(),
+                                    nullable=True))
+
+
+def downgrade():
+    op.drop_column('runs', 'lock_timestamp')
+
+    # If there are runs who are in the database with '-2' duration, they
+    # must be removed as these rows signify runs which store operation didn't
+    # conclude.
+    connection = op.get_bind()
```

Comment:
A cleanup for stale runs with the same logic exists in the database cleanup too.

This is the **down**grade script. In case of a downgrade to an older version of CodeChecker, the cleanup *MUST* happen, because CodeChecker versions released prior to merging this feature were not equipped to handle `-2` in `duration` properly.